hack/generated/pkg/reconcilers/azure_deployment_reconciler.go

```diff
@@ -109,18 +109,34 @@ func NewAzureDeploymentReconciler(
 	}
 }
 
+func (r *AzureDeploymentReconciler) checkForFatalReconciliationError(err error) {
+	var fatal FatalReconciliationError
+	if errors.As(err, &fatal) {
+		r.SetResourceProvisioningState(armclient.FailedProvisioningState)
+		r.SetResourceError(fatal.Message)
+	}
+}
+
 func (r *AzureDeploymentReconciler) CreateOrUpdate(ctx context.Context) (ctrl.Result, error) {
 	action, actionFunc, err := r.DetermineCreateOrUpdateAction()
 	if err != nil {
 		r.log.Error(err, "error determining create or update action")
 		r.recorder.Event(r.obj, v1.EventTypeWarning, "DetermineCreateOrUpdateActionError", err.Error())
+
+		r.checkForFatalReconciliationError(err)
+
 		return ctrl.Result{}, err
 	}
 
+	r.log.Info("Reconciling Azure resource %s, will perform: %s", r.obj.AzureName(), action)
```

Comment:
Is this equivalent to `V(0)`?

Reply:
I think there might not be any `V`s on this type, but will check